modules/core/src/helpers/access-control.ts

```diff
@@ -79,23 +112,35 @@ export const hasRequiredScopes = (
  * @remarks
  * Currently the check passes if at least one feature has the required read permissions.
  *
- * @param {FeatureAccessConfigInterface} featureConfig - Feature configuration.
+ * @param featureConfig - `FeatureAccessConfigInterface` Feature configuration.
+ * @param organzationType - `string` Organization type. This should be equals to the `OrganizationType` enum in
+ * `modules/common/src/constants/organization-constants.ts`.
+ * @param isLegacyRuntimeDisabled - `boolean` Is legacy runtime disabled. This is used to ensure backward compatibility.
  *
- * @return {boolean} True is access is granted, false if not.
+ * @returns `boolean` True is access is granted, false if not.
  */
-export const isPortalAccessGranted = <T = {}>(featureConfig: T, allowedScopes: string): boolean => {
-    const isDefined = featureConfig && !isEmpty(featureConfig);
+export const isPortalAccessGranted = <T = unknown>(
+    featureConfig: T, 
+    allowedScopes: string, 
+    organzationType?: string, 
+    isLegacyRuntimeDisabled?: boolean
+): boolean => {
+    const isDefined: boolean = featureConfig && !isEmpty(featureConfig);
```

Comment:
Do we need a separate constant for this?

Reply:
Did this for readability